src/api/XRFrame.js

```diff
@@ -78,18 +88,18 @@ export default class XRFrame {
     if (!this[PRIVATE].active) {
       throw new DOMException(NON_ACTIVE_MSG, 'InvalidStateError');
     }
-    if (space._specialType === "viewer") {
-      // Don't just return the viewer pose since the resulting pose shouldn't
-      // include the views array - it should just have the transform.
-      let viewerPose = this.getViewerPose(baseSpace);
-      return new XRPose(
-        new XRRigidTransform(viewerPose.poseModelMatrix),
-        viewerPose.emulatedPosition);
-    }
 
+    const device = this[PRIVATE].device;
     if (space._specialType === "target-ray" || space._specialType === "grip") {
-      return this[PRIVATE].device.getInputPose(
+      // TODO: Stop special-casing input.
+      return device.getInputPose(
         space._inputSource, baseSpace, space._specialType);
+    } else {
+      space._ensurePoseUpdated(device, this[PRIVATE].id);
+      baseSpace._ensurePoseUpdated(device, this[PRIVATE].id);
```

Comment:
Does this also need to be called before line 96?

Reply:
Eventually yes, but that's using an older path for now that computes the poses via a completely different path so they don't need the same update call. I'm going to follow this PR with one specifically to refactor that bit to use the same logic as everything else soon, but I was already wary of how large this change was and didn't want to complicate it further.